extensions/notebook/src/jupyter/jupyterServerManager.ts

```diff
@@ -108,9 +108,11 @@ export class LocalJupyterServerManager implements nb.ServerManager, vscode.Dispo
 
 	private async doStartServer(kernelSpec: nb.IKernelSpec): Promise<IServerInstance> { // We can't find or create servers until the installation is complete
 		let installation = this.options.jupyterInstallation;
-		await installation.promptForPythonInstall(kernelSpec.display_name);
-		if (!installation.previewFeaturesEnabled) {
-			await installation.promptForPackageUpgrade(kernelSpec.display_name);
+		if (!installation.runningOnSAW) {
```

Comment:
I would expect that if python is already installed (which I assume it is for the SAW) that the "prompt" would just be a no-op. Is that not the case?

Reply:
We have to use python to check if our required packages are installed. If some things are missing, then we open the prompt. However, I can move the runningOnSaw check into the python methods to reduce checks elsewhere.

Reply:
I'm not sure I understand what the problem is. Shouldn't the python already be installed and have all the required packages?

Reply:
Correct, which is why we completely skip checking for it here. I've already fixed the methods here to be no-ops when running on a SAW, instead of checking it where the methods are being used elsewhere.

Reply:
So you're just doing this as an optimization? Because shouldn't these calls end up being no-ops since everything is installed already?

Are we at all concerned about the saw environment being changed and the required packages not being there anymore? What if we did the checks and then if they failed on saw we just notify the user that the python installation is missing some required dependencies?

Reply:
These calls are already no-ops in the most recent version of these changes. They just return immediately when running on a SAW.

Installing pip packages is blocked on SAWs, so if a user starts uninstalling required packages the only way to fix it would be a complete reinstall of ADS. Plus modifying the package detection logic increases the risk from this change, which might block it from getting into the release.

Reply:
We also disable Manage Packages on SAWs, so if they uninstall packages it's because they're modifying the python environment outside of ADS, which would be like deleting DLLs from an app's install directory.

Reply:
Yeah - I'm more coming from the side of "if something happened to the python environment then it would be nice to catch it earlier rather than later when the user tries to run a Notebook". Might help prevent issues if for example the saw builds change what packages they're installing for some reason and thorough testing isn't done.

Anyways not a huge deal to leave it like this - I'm just trying to push for having these kind of super specific checks in random places in the code if we could make it have the same experience without them.